src/compiler/core.rs

```diff
@@ -1,13 +1,42 @@
-pub fn for_each<TItem, TReturn>(
-    array: &[TItem],
-    callback: &mut dyn FnMut(&TItem, usize) -> Option<TReturn>,
+use crate::SortedArray;
+
+pub fn for_each<
+    TCollection: IntoIterator,
```

Comment:
I think I changed this so that I can pass a `NodeArray` to `for_each()`? But I don't remember